core/src/consensus.rs

```diff
@@ -208,6 +208,10 @@ pub const AR_SCALE_DAMP_FACTOR: u64 = 13;
 pub fn graph_weight(height: u64, edge_bits: u8) -> u64 {
 	let mut xpr_edge_bits = edge_bits as u64;
 
+	if edge_bits < global::base_edge_bits() {
+		return 0;
+	}
```

Comment:
What is this for?

Reply:
It adds range checking.

Reply:
I can do the check in the caller if you prefer?

Reply:
I think you can remove it altogether. Correct me if I'm wrong but, we don't need this check in the caller since it's only called for a `valid_share`.

Reply:
Its called when building a new block template.  That includes the first new block template before any shares are submitted.
Since we set a default size in run_loop() you are correct that the current code never calls with an invalid value - but maybe it would be best to have the check to futureproof a mistake?   I dont feel strongly so I'll remove the check as suggested.